file.go

```diff
@@ -0,0 +1,134 @@
+package hercules
+
+import "fmt"
+
+type File struct {
+	tree   *RBTree
+	status map[int]int64
+}
+
+const TreeEnd int = -1
+
+func min(a int, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func max(a int, b int) int {
+	if a < b {
+		return b
+	}
+	return a
+}
+
+func (file *File) Initialize(time int, length int, status map[int]int64) {
```

Comment:
This looks as a ctor, in Go we usually name them `New(time int, length int, status map[int]int64)` instead.